### PR TITLE
chore: update go and golanci lint

### PR DESCRIPTION
**Pull request by ramonpetgrave64**
This PR updates go to 1.23.1 and updates golanci-lint to v1.61.1, while fixing new lint errors.

### CODE REVIEW

**Comment by ramonpetgrave64** [comment]
@loosebazooka

File: .golangci.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ run:
   concurrency: 2
   deadline: 5m
   # For generics.
-  go: 1.18
+  # go: 1.23.1
```
[the remark this answers is not in the document]
Comment (jku) [reply .golangci.yml:6]: maybe no need to leave version number in here, it won't stay updated

**Review by jku — APPROVED** [review]

**Review by loosebazooka — APPROVED** [review]
+1 to jku's comment, but LGTM

**Comment by ramonpetgrave64** [comment]
> +1 to jku's comment, but LGTM

removed, please +1 again.

**Review by loosebazooka — APPROVED** [review]

**Pull request merged**